packages/admin/admin/src/common/toolbar/ToolbarBreadcrumbs.tsx

```diff
@@ -0,0 +1,295 @@
+                    component: "div",
+                    sx: {
+                        paddingTop: 3,
+                        paddingBottom: 3,
+                    },
+                }}
+                PaperProps={{
+                    sx: {
+                        width: menuWidth,
+                    },
+                }}
+            >
+                {breadcrumbs.map(({ title, url }, nestingLevel) => {
+                    const isCurrentPage = nestingLevel === breadcrumbs.length - 1;
+
+                    return (
+                        <MobileMenuItem key={nestingLevel} onClick={toggleMobileMenu} selected={isCurrentPage} to={url} component={BreadcrumbLink}>
+                            {nestingLevel > 0 && <MenuItemNestingIndicator nestingLevel={nestingLevel} />}
+                            <ListItemText
+                                primary={title}
+                                primaryTypographyProps={{
+                                    sx: {
+                                        fontWeight: isCurrentPage ? 600 : 300,
+                                        textOverflow: "ellipsis",
+                                        overflow: "hidden",
+                                    },
+                                }}
+                            />
+                        </MobileMenuItem>
+                    );
+                })}
+            </Menu>
+        </>
+    );
+};
+
+const Root = styled("div")(
+    ({ theme }) => css`
+        display: flex;
+        align-items: center;
+        gap: ${theme.spacing(2)};
+        padding-left: ${theme.spacing(2)};
+        padding-right: ${theme.spacing(2)};
+    `,
+);
+
+const DesktopBreadcrumbs = styled("div")(
+    ({ theme }) => css`
+        display: none;
+        align-items: center;
+        gap: ${theme.spacing(1)};
+        overflow: hidden;
+
+        ${theme.breakpoints.up("md")} {
+            display: flex;
+        }
+    `,
+);
+
+const MobileBreadrumbs = styled(ButtonBase)(
+    ({ theme }) => css`
+        display: flex;
+        align-items: center;
+        gap: ${theme.spacing(1)};
+        flex-grow: 1;
+        overflow: hidden;
+        padding-top: ${theme.spacing(2)};
+        padding-bottom: ${theme.spacing(2)};
+
+        ${theme.breakpoints.up("md")} {
+            display: none;
+        }
+    `,
+);
+
+const getCommonItemStyles = (theme: Theme) => css`
+    color: ${theme.palette.grey[900]};
+    text-overflow: ellipsis;
+    white-space: nowrap;
+    overflow: hidden;
+
+    ${theme.breakpoints.up("md")} {
+        padding-top: ${theme.spacing(2)};
+        padding-bottom: ${theme.spacing(2)};
+    }
+`;
+
+const Item = styled(Typography)(
+    ({ theme }) => css`
+        ${getCommonItemStyles(theme)}
+
+        ${theme.breakpoints.up("md")} {
+            display: block;
+        }
+    `,
+) as typeof Typography;
```

Comment:
Why is this needed? 🤔

Reply:
Because of this issue: https://mui.com/material-ui/guides/typescript/#complications-with-the-component-prop
We have this problem in multiple existing components.